lib/cli/daemoncommand.cpp

```diff
@@ -191,6 +191,37 @@ pid_t l_UmbrellaPid = 0;
 static Atomic<bool> l_AllowedToWork (false);
 #endif /* _WIN32 */
 
+#ifdef I2_DEBUG
+/**
+ * Determine whether the developer wants to delay the worker process to attach a debugger to it.
+ *
+ * @return Internal.DebugWorkerDelay double
+ */
+static double GetDebugWorkerDelayCached()
+{
+	static double debugWorkerDelay = -1;
+
+	if (debugWorkerDelay != -1)
+		return debugWorkerDelay;
+
+	debugWorkerDelay = 0.0;
+
+	Namespace::Ptr internal = ScriptGlobal::Get("Internal", &Empty);
+
+	if (!internal)
+		return false;
+
+	Value vdebug;
+
+	if (!internal->Get("DebugWorkerDelay", &vdebug))
+		return false;
+
+	debugWorkerDelay = Convert::ToDouble(vdebug);
+
+	return debugWorkerDelay;
```

Comment:
IMAO it would be more readable to do an if-uninitialized-then-initialize and a single return after the if.

Reply:
I just copied what we do with Internal.DebugJsonRpc, did not yet consider optimizations.